File: frontend/src/api/job.ts
import type { AxiosResponse } from 'axios';
import request from '@/utils/request';
import type {
  Job,
  JobType,
  JobStatus,
  JobShellParams,
  JobHTTPParams,
  JobDataXParams,
  CreateJobRequest,
  UpdateJobRequest,
  JobListRequest,
  JobListResponse,
  JobHistory,
  JobHistoryListRequest,
  JobHistoryListResponse,
} from './types';

export type {
  Job,
  JobType,
  JobStatus,
  JobShellParams,
  JobHTTPParams,
  JobDataXParams,
  CreateJobRequest,
  UpdateJobRequest,
  JobListRequest,
  JobListResponse,
  JobHistory,
  JobHistoryListRequest,
  JobHistoryListResponse,
};

// 获取任务列表
export function getJobList(params: JobListRequest): Promise<AxiosResponse<JobListResponse>> {
  return request.get('/jobs', { params });
}

// 创建任务
export function createJob(data: CreateJobRequest): Promise<AxiosResponse<Job>> {
  console.log(data);
  return request.post('/jobs', data);
}

// 更新任务
export function updateJob(id: number, data: UpdateJobRequest): Promise<AxiosResponse<Job>> {
  return request.put(`/jobs/${id}`, data);
}

// 删除任务
export function deleteJob(id: number): Promise<AxiosResponse<void>> {
  return request.delete(`/jobs/${id}`);
}

// 启动任务
export function startJob(id: number): Promise<AxiosResponse<void>> {
  return request.post(`/jobs/${id}/start`);
}

// 停止任务
export function stopJob(id: number): Promise<AxiosResponse<void>> {
  return request.post(`/jobs/${id}/stop`);
}

// 执行一次任务
export function executeJob(id: number): Promise<AxiosResponse<void>> {
  return request.post(`/jobs/${id}/execute`);
}

// 批量执行任务
export function executeJobs(ids: number[]): Promise<AxiosResponse<void>> {
  return request.post('/jobs/execute', { ids });
}

// 获取任务执行历史
export function getJobHistoryList(params: JobHistoryListRequest): Promise<AxiosResponse<JobHistoryListResponse>> {
  return request.get('/jobs/history', { params });
}

// 清理任务历史
export function cleanJobHistory(days: number): Promise<AxiosResponse<void>> {
  return request.post('/jobs/history/clean', { days });
}

File: routes/api/v1/job.go
package v1

import (
	"datax-admin/controllers"
	"github.com/gin-gonic/gin"
)

// RegisterJobRoutes 注册任务管理相关路由
func RegisterJobRoutes(authenticated *gin.RouterGroup) {
	jobController := controllers.NewJobController()
	
	authenticated.POST("/jobs", jobController.CreateJob)
	authenticated.PUT("/jobs/:id", jobController.UpdateJob)
	authenticated.DELETE("/jobs/:id", jobController.DeleteJob)
	authenticated.POST("/jobs/:id/start", jobController.StartJob)
	authenticated.POST("/jobs/:id/stop", jobController.StopJob)
	authenticated.POST("/jobs/:id/execute", jobController.ExecuteJob)
	authenticated.POST("/jobs/execute", jobController.ExecuteJobs)
	authenticated.GET("/jobs", jobController.GetJobList)
	authenticated.GET("/jobs/history", jobController.GetJobHistoryList)
	authenticated.POST("/jobs/history/clean", jobController.CleanJobHistory)
}